### ISSUE TITLE
Мониторинг выделения памяти

### ISSUE THREAD

**Issue by mamontov-cpp** [permyakov-i/cppjudge#4]
Выделение памяти должно мониториться во время исполнения, а не после. Надо как-то начать это обрабатывать.

**Comment by permyakov-i** [comment]
6ccc9a99341a6977f8315a79343011a9018ba7aa

**Comment by mamontov-cpp** [comment]
Нет, этого недостаточно. Вы получаете значение памяти один раз, проверяете и убиваете. 

Процесс может выделять память и после вашей проверки. Вы должны это делать все время, пока процесс исполняется.

**Comment by permyakov-i** [comment]
Сделано.

**Comment by mamontov-cpp** [comment]
Прочитайте внимательно свой код в  https://github.com/permyakov-i/cppjudge/commit/86ca1e4a16c6ab93949261cb0c173a18d38c12d2

По-моему  } while (!proc.WaitForExit(timeLimit * 1000)); довольно неудачное условие для проверки.
Просто добавьте в репозиторий тестовый пример процесса, который выделяет память ЗНАЧИТЕЛЬНО больше лимита и не завершается - будет  ли ваша программа работать с ним корректно?

**Comment by permyakov-i** [comment]
А смысл ждать дольше, если время выполнения программы уже превышено? Проверяю память пока не истечет время, потом убиваю процесс по таймауту.

**Comment by mamontov-cpp** [comment]
Просто ответьте на вопрос - сколько раз за время работы выполнится проверка памяти на превышение?

**Comment by permyakov-i** [comment]
Столько итераций, сколько успеет пройти цикл проверки пока либо программа не завершится, либо не истечет таймаут. Проверка, завершился ли процесс по флагу hasExited находится внутри цикла. В качестве подтверждения пример из mdsn: https://msdn.microsoft.com/ru-ru/library/system.diagnostics.process.virtualmemorysize64(v=vs.110).aspx

**Comment by permyakov-i** [comment]
Хотя, я так подумал, можно и правда проверять по свойству Exited.

**Comment by permyakov-i** [comment]
А собственно у меня и эта проверка уже есть, внутри цикла, совсем забыл про нее. И да, программа работает корректно в таком случае.

**Comment by mamontov-cpp** [comment]
А вот и нет. Чтобы проверить достаточно запустить абсолютно несложную программу на вашем тестере (папка 7 - тесты):

https://drive.google.com/file/d/1DXZUvelCWdP9LAoNsJvuoVAPOtuzD7-X/view?usp=sharing

Данная программа спокойно может съесть у меня 1,5Гб памяти, после чего упасть с исключением. Реальная программа студента -  может и не упасть. Оба варианта недопустимы - в идеале программа должна быть завершена когда наступило условие выхода за ограничения по памяти.

Я не говорю о том, что зачем-то мне выдается message box  c исключением, который вообще не нужен. Кстати вы пробовали программу позапускать несколько раз на одном и том же тесте?
У вас почему-то отчетные списки растут. Это тоже надо исправить.

**Comment by permyakov-i** [comment]
Изменил условие, теперь убивается более стабильно. (Ваш тестовый пример тоже)

**Comment by mamontov-cpp** [comment]
Спасибо. Исправьте ещё два замечания выше.

**Comment by permyakov-i** [comment]
Про месседж боксы я понял, а про отчетные списки не очень - растет содержимое окна Результаты тестирования?

**Comment by mamontov-cpp** [comment]
Да. Там вывод выглядит наподобие:
Test 0: Fail
Test 1: Fail
Test 0: Fail
Test 1: Fail
... и. т. д.
Если прогонять одну и ту же программу несколько раз на одном и том же тестовом наборе

**Comment by permyakov-i** [comment]
Исправил.
